ci/pipelines/deploy-smoke-tests.yml

```diff
@@ -92,6 +92,12 @@ jobs:
         params:
           <<: *aws_assumed_role_creds
           ENVIRONMENT: test
+      - file: pay-ci/ci/tasks/find-terraform-version.yml
+        params:
+          TERRAFORM_ROOT: pay-infra/provisioning/terraform/deployments/deploy/deploy-7/management/smoke_tests
```

Comment:
I'm just wondering whether this could be made simpler and potentially less error prone, since we only have a single `.terraform-version` file in pay-infra, we don't really need to pass the root deployment (which is where I can see the human error creeping in and which I began looking at closely to check for some error). Looking at the script it will just use the root of pay-infra if the `TERRAFORM_ROOT` is not set https://github.com/alphagov/pay-ci/blob/master/ci/scripts/find-terraform-version.sh#L6 and that's the only place we have the `.terraform-version` set. Unless we want to keep passing the dir in case we wish to diverge versions per deployment in future, but we could set it as and when needed??

Reply:
I'd rather keep the behaviour that tfenv has, it's a lot easier to explain that it works the same as the other tooling in the tf ecosystem than to have our own special case.

I'm planning to refactor (not now, but in firebreak) the actual code which does the terraform deployments to be a lot more generalised than it is now (7 or 8 different sets of code which only varies the args passed to terraform, the wait-for-deploy task is already generalised like this), and that would need the terraform root anyway.